tests/api/test_core.py

```diff
@@ -288,6 +289,68 @@ def _run_test():
     assert exc.value.args[0] == "Permissions should be a tuple or a string in Meta"
 
 
+MUTATION_CREATE_TOKEN = """
+    mutation tokenCreate($email: String!, $password: String!){
+        tokenCreate(email: $email, password: $password) {
+            token
+            user {
+                email
+            }
+            errors {
+                field
+                message
+            }
+            accountErrors {
+                field
+                message
+                code
+            }
+        }
+    }
+"""
+
+
+def test_create_token(api_client, customer_user):
+    variables = {"email": customer_user.email, "password": customer_user._password}
+    response = api_client.post_graphql(MUTATION_CREATE_TOKEN, variables)
+    content = get_graphql_content(response)
+    user_email = content["data"]["tokenCreate"]["user"]["email"]
+    assert customer_user.email == user_email
+    assert content["data"]["tokenCreate"]["token"]
+    assert content["data"]["tokenCreate"]["errors"] == []
+    assert content["data"]["tokenCreate"]["accountErrors"] == []
+
+
+def test_create_token_invalid_password(api_client, customer_user):
+    variables = {"email": customer_user.email, "password": "wrongpassword"}
+    response = api_client.post_graphql(MUTATION_CREATE_TOKEN, variables)
+    content = get_graphql_content(response)
+    assert (
+        content["data"]["tokenCreate"]["errors"][0]["message"]
+        == "Please, enter valid credentials"
+    )
+    assert (
+        content["data"]["tokenCreate"]["accountErrors"][0]["code"]
+        == AccountErrorCode.INVALID_CREDENTIALS.value.upper()
+    )
+    assert content["data"]["tokenCreate"]["accountErrors"][0]["field"] == "email"
```

Comment:
We shouldn't check messages form errors
```suggestion
    error = content["data"]["tokenCreate"]["errors"][0]
    assert error["code"] == AccountErrorCode.INVALID_CREDENTIALS.value.upper()
    assert error["field"] == "email"
```

Reply:
This is a check fo error code. We can check if it is not empty. Not sure what is the preferred option here.

Reply:
We prefer checking `code` and `field` these fields should provide enough information to display error on storefronts/dashboard. Messages are information for developers. We should have possibility to change the message without changing tests.

In operation with return params in errors, we should check params too. For example `ProductVariantBulkCreate` contains `index` in error and this field needs to be checked in tests.

Reply:
That's what I am checking here. AccountErrorCode and field.

Reply:
Ow damn. Yes I left the message...